Src/Base/AMReX_ParallelDescriptor.H

```diff
@@ -1005,6 +1013,79 @@ ParallelDescriptor::Gatherv (const T* send, Long sc,
     BL_COMM_PROFILE(BLProfiler::Gatherv, std::accumulate(rc.begin(),rc.end(),0L)*sizeof(T), root, BLProfiler::NoTag());
 }
 
+template <class T>
+void
+ParallelDescriptor::GatherLayoutDataToVector (const LayoutData<T>& sendbuf,
+                                              Vector<T>& recvbuf, int root)
+{
+    BL_PROFILE_T_S("ParallelDescriptor::GatherLayoutData(Ti)", T);
+    
+#ifdef BL_USE_MPI
+    // Gather prelims
+    Vector<int> index_to_send = sendbuf.IndexArray();
+    Vector<T> T_to_send;
+    T_to_send.reserve(sendbuf.local_size());
+    
+    for (auto i : sendbuf.IndexArray())
+    {
+        T_to_send.push_back(sendbuf[i]);
+    }
+
+    int nprocs = ParallelContext::NProcsSub();
+    Vector<Long> recvcount(nprocs), disp(nprocs);
+
+    const Vector<int>& old_pmap = sendbuf.DistributionMap().ProcessorMap();
+    for (int rank : old_pmap)
+    {
+        ++recvcount[rank];
+    }
+
+    std::partial_sum(recvcount.begin(), --recvcount.end(), ++disp.begin());
+    Vector<T> new_index_to_T(sendbuf.size());
+
+    ParallelDescriptor::Gatherv(&T_to_send[0],
+                                T_to_send.size(),
+                                &new_index_to_T[0],
+                                recvcount,
+                                disp,
+                                root);
+
+    Vector<int> new_index_to_old_index(sendbuf.size());
+    std::iota(new_index_to_old_index.begin(), new_index_to_old_index.end(), 0);
+    
+    // stable_sort to prevent reorderings of index with same rank
+    std::stable_sort(new_index_to_old_index.begin(), new_index_to_old_index.end(),
```

Comment:
Why do we need to sort?

Reply:
To clarify, I understand how you do this with sort.  However, we could avoid sort because it's expensive.  Like in your previous implementation, we could instead build `Vector<Vector<int> >`, where the outer vector has the size of `nprocs`.  When we loop over `old_pmap`, we can build a vector like `IndexArray()` for each process.

Reply:
Thanks for catching this again, that is a better way to do it.